plugins/woocommerce/src/Internal/DataStores/Orders/OrdersTableDataStore.php

```diff
@@ -773,6 +796,92 @@ private function generate_select_clause_for_props( $table_alias, $props ) {
 		return implode( ', ', $select_clauses );
 	}
 
+	/**
+	 * Persists order changes to the database.
+	 *
+	 * @param \WC_Order $order        The order.
+	 * @param boolean   $only_changes Whether to persist all order data or just changes in the object.
+	 * @return void
+	 */
+	protected function persist_order_to_db( $order, $only_changes = true ) {
+		global $wpdb;
+
+		// XXX implement case $only_changes = false.
+		$changes = $only_changes ? $order->get_changes() : array();
+
+		// Figure out what needs to be updated in the database.
+		$db_updates = array();
+
+		// wc_orders.
+		$row = $this->helper->get_db_row_from_order_changes( $changes, $this->order_column_mapping );
+		if ( $row ) {
+			$db_updates[] = array_merge(
+				array(
+					'table'        => self::get_orders_table_name(),
+					'where'        => array( 'id' => $order->get_id() ),
+					'where_format' => '%d',
+				),
+				$row
+			);
+		}
+
+		// wc_order_operational_data.
+		$row = $this->helper->get_db_row_from_order_changes(
```

Comment:
> you can get the current DB details from get_order_data_for_ids method, which will return a DB object for the current order, which you can then use to compare if you want to.

I was using the order object itself as "source of truth" particularly as that's how the other datastores work and doesn't involve another SELECT. Also because it could be used for the `create()` method, which doesn't have anything to compare against already in the db.

Reply:
> I wonder if we were to implement locking sometime in the future, should we add a provision in the DB schema itself right now? Maybe at least an edit lock column.

I think that makes sense. I've opened issue #33173 for that, so that we don't forget.